x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/Options.java

```diff
@@ -112,4 +151,54 @@ public static void populateMap(
             }
         }
     }
+
+    public static void populateMapWithExpressionsMultipleDataTypesAllowed(
```

Comment:
Added this method as it's way easier in `Decay` to work with `Expression` objects, which have a `DataType` associated with them. I assume it would've probably been wrong/dangerous to rely on DataType.convert(...) by trying out a sequence of allowed data types.